### PR TITLE
Adding background session capabilities to file/folder restore item

### PR DESCRIPTION
**Pull request by jdizoglio**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.h [hunk 1]
```diff
@@ -7,6 +7,18 @@
 
 @interface BOXTrashedFileRestoreRequest : BOXRequest
 
+// NOTE: Both the associateID and requestDirectoryPath values are required for performing the request in the background.
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.h:10]: Maybe add `This is required…` note to `requestDirectoryPath`

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.m [hunk 2]
```diff
@@ -41,49 +51,120 @@ - (BOXAPIOperation *)createOperation
...
+        __weak BOXAPIDataOperation *weakFileOperation = fileOperation;
+        
+        fileOperation.successBlock = ^(NSString *modelID, long long expectedTotalBytes) {
+            NSData *data = [NSData dataWithContentsOfFile:weakFileOperation.destinationPath];
+            BOXFile *file = nil;
+            if (data != nil) {
+                NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
+                file = (BOXFile *)[[self class] itemWithJSON:jsonDictionary];
+                
+            if ([self.cacheClient respondsToSelector:@selector(cacheTrashedFileRestoreRequest:withFile:error:)]) {
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.m]: Indentation

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFolderRestoreRequest.h [hunk 3]
```diff
@@ -7,6 +7,18 @@
 
 @interface BOXTrashedFolderRestoreRequest : BOXRequest
 
+// NOTE: Both the associateID and requestDirectoryPath values are required for performing the request in the background.
```
Comment (jlawton) [inline BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFolderRestoreRequest.h:10]: Maybe add `This is required…` note to `requestDirectoryPath`

**Review by jlawton — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.m (hunk 2 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.m]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFolderRestoreRequest.h (hunk 3 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFolderRestoreRequest.h:10]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.h (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXTrashedFileRestoreRequest.h:10]: done

**Pull request merged**